cmd/esc/cli/env_set.go

```diff
@@ -59,7 +59,15 @@ func newEnvSetCmd(env *envCommand) *cobra.Command {
 
 			var yamlValue yaml.Node
 			if err := yaml.Unmarshal([]byte(args[1]), &yamlValue); err != nil {
-				return fmt.Errorf("invalid value: %w", err)
+				// Default to a string value
+				yamlValue = yaml.Node{
```

Comment:
@pgavlin are there any cases where we **don't** want to default to the string value if the given value can't be parsed in YAML?

Reply:
My gut is to say "all cases, as this defaulting makes the behavior harder to understand". I can imagine adding a flag (e.g. `--string`) that forces the ESC CLI to treat its input as a string and not attempt to parse it as YAML at all.

Reply:
something like this: https://github.com/pulumi/esc/compare/pgavlin/env-set-string

Reply:
@pgavlin i made a [PR](https://github.com/pulumi/esc/pull/467) out of your branch